tracer/src/Datadog.Trace/TraceId.cs

```diff
@@ -84,6 +84,14 @@ public int CompareTo(object? obj)
     [Pure]
     public override int GetHashCode() => HashCode.Combine(Lower, Upper);
 
+    [Pure]
+    public string ToUpperHexString()
+    {
+        return this == Zero ?
+                   "0000000000000000" :
+                   HexString.ToHexString(Upper);
+    }
+
```

Comment:
:kill-it-with-fire:
```suggestion
```

Reply:
Done in 5879a4e6f33dc6e8fc21d3bf2e03cdf01eeb20d1